Tasks/CordovaBuild/node-setup.js

```diff
@@ -4,32 +4,52 @@
 */
 
 var path = require('path'),
+    fs = require('fs'),
     semver = require('semver'),
     nodeManager = require('./lib/node-manager.js'),
     taskLibrary = require('./lib/vso-task-lib-proxy.js'),
     teambuild = require('taco-team-build');
 
 var nodeSetupPromise;
+
+var cwd = taskLibrary.getInput('cwd', /* required */ false);
+var tacoFile = path.join((cwd ? cwd : '.'), 'taco.json');
+
 var version = taskLibrary.getInput('cordovaVersion', /* required */ false);
+if (!version) {
+    try {
+        var stats = fs.statSync(tacoFile);
+        if (stats && stats.isDirectory()) {
+            version = require(tacoFile)['cordova-cli'];
+            console.log('Cordova version set to ' + version + ' based on the contents of taco.json');
+        }
+    } catch (e) { }
+}
+
```

Comment:
I thought the general format for the env var was PACKAGENAMEALLCAP_DEFAULT_VERSION, so we would only need to check for CORDOVA_DEFAULT_VERSION in this case? or does the version of the other packages imply a certain cordova version? if that is the case, there needs to be additional logic here to convert the CLI version into a corresponding cordova version. The ionic and phonegap version numbers do not seem to line up with the Cordova ones.